RocketChat/Rocket.Chat
# fix: Correct email validation and error message display in MailExportForm

## Proposed changes (including videos or screenshots)
The validateEmail function has been updated to correctly validate email addresses according to the RFC standard. It now correctly identifies email addresses without a top-level domain (e.g., "user@domain") as invalid.

The error message for the messagesCount field is now only displayed when the user doesn't select any message
https://github.com/user-attachments/assets/5c0fa949-ff12-4b41-9718-c5b226d7c82c




## Issue(s)
#32974 

## Steps to test or reproduce
1.Navigate to the MailExportForm component.
2.Test the email validation by entering an email address without a top-level domain and attempting to submit the form. The form should correctly identify the email as invalid.
3.Select a message. The error message for the messagesCount field should not be displayed when you have selected the messages.

## Further comments
